### PR TITLE
Apm control little cleanup and refactoring

### PR DESCRIPTION
**Pull request by khancyr**
remove tabs, and fix some spaces and trailling withspaces
remove unimplemented functions
remove unused parameter
use c++ cast and fix implicit cast
correct abs usage
add f qualifier for float
use is_positive when possible
make variable const when possible, make some function from AHRS and dataflash const

### CODE REVIEW

**Comment by tridge** [comment]
Is this tested? How much time have you spent testing this in SITL plane?

**Comment by khancyr** [comment]
Tested only on SITL and autotest. I don't have plane nor copter

File: libraries/APM_Control/AP_SteerController.cpp [hunk 1]
```diff
@@ -199,7 +199,7 @@ int32_t AP_SteerController::get_steering_out_rate(float desired_rate)
     }
 
     // Convert to centi-degrees and constrain
-    return constrain_float(_last_out * 100, -derate_constraint, derate_constraint);
+    return static_cast<int32_t>(constrain_float(_last_out * 100.0f, -derate_constraint, derate_constraint));
```
Comment (peterbarker) [inline libraries/APM_Control/AP_SteerController.cpp]: I don't think this counts as a "little cleanup" any more.

**Comment by rmackay9** [comment]
I don't know what to do about this PR in total, but I've pulled out the changes to APM_Control/AR_AttitudeControl and pushed to master.

**Comment by rmackay9** [comment]
I'm going to close this because I've pulled in the bits from areas that I'm responsible for.  I think there are others that could go in as well including the dataflash and the ahrs.wind_estimate change but that's outside my jurisdiction.

**Pull request closed without merging**